test/03_USM.test.js

```diff
@@ -29,127 +43,280 @@ contract('USM', (accounts) => {
+          const ethPool2 = (await weth.balanceOf(usm.address))
+          ethPool2.toString().should.equal(oneEth.mul(TWO).toString())
+
+          const usmBalance2 = (await usm.balanceOf(user2))
+          // Verify that sliding prices are working: this mirrors the sliding FUM prices above - see the math in USM.usmFromMint(),
+          // with mint() here 2xing the pool (1 ETH from fund() earlier -> 2) and the initial USM price of $1 = 0.004 ETH, the USM
+          // added should be 1 (current ETH pool) / 0.004 * (1 - 1 / 2) = 125 (ie, the ETH price $250 / 2):
+          usmBalance2.toString().should.equal(wadMul(oneEth, priceWAD).div(TWO).toString())
+
+          const mintBurnAdj2 = (await usm.mintBurnAdjustment())
+          // We've 2x'd the ETH pool, which according to the adjustment math in USM.mint(), should 0.25x the mintBurnAdjustment:
+          mintBurnAdj2.toString().should.equal(WAD.div(FOUR).toString())
+
+          const fumBuyPrice2 = (await usm.fumPrice(sides.BUY))
+          const fumSellPrice2 = (await usm.fumPrice(sides.SELL))
+          // FUM buy price should be unaffected (fundamentally, minting doesn't change the FUM price); but FUM sell price should
+          // now have mintBurnAdjustment applied (since both adjustments apply to both USM and FUM price!), ie, be 1/4 buy price.
+          // (Recall that mint() affects FUM *sell* price because minting (buying USM) and selling FUM are both "short ETH" ops.)
+          fumSellPrice2.toString().should.equal(wadMul(fumBuyPrice2, mintBurnAdj2).toString())
+
+          // Check that mintBurnAdjustment decays properly (well, approximately) over time:
+          // - Our adjustment was previously 1/4: see mintBurnAdj2 above.
+          // - Use timeMachine to move forward 180 secs = 3 min.
+          // - Since USM.BUY_SELL_ADJUSTMENTS_HALF_LIFE = 60 (1 min), 3 min should mean a decay of ~1/8.
+          // - The way our "time decay towards 1" approximation works, decaying 1/4 by ~1/8 should yield 1 + (1/4 * 1/8) - 1/8 = 0.90625.
+          //   (See comment in USM.mintBurnAdjustment().)
+          // - And finally - since our decay's exponentiation is approximate (WadMath.wadHalfExp()), don't compare mintBurnAdj3 to
+          //   targetMintBurnAdj3 *exactly*, but instead first round both to the nearest multiple of 10 (via divRound(10)).
+          const block0 = (await web3.eth.getBlockNumber())
+          const t0 = (await web3.eth.getBlock(block0)).timestamp
+          const timeDelay = 3 * 60
+          await timeMachine.advanceTimeAndBlock(timeDelay)
+          const block1 = (await web3.eth.getBlockNumber())
+          const t1 = (await web3.eth.getBlock(block1)).timestamp
+          t1.toString().should.equal((t0 + timeDelay).toString())
+          const mintBurnAdj3 = (await usm.mintBurnAdjustment())
+          const decayFactor3 = WAD.div(EIGHT)
+          const targetMintBurnAdj3 = WAD.add(wadMul(mintBurnAdj2, decayFactor3)).sub(decayFactor3)
```

Comment:
Should we extract this into a helper function?

Reply:
Done